figures/mapping-histology-labels.Rmd

```diff
@@ -56,284 +46,340 @@ Rscript -e "rmarkdown::render('figures/mapping-histology-labels.Rmd',
+These are best used in conjunction with _labels_ in figures, but are intended to allow readers to "track" labels _across figures_.
 
-Print out the number of `display_group` (including `normal`)!
+Where there's a 1:1 mapping between `broad_histology` and `cancer_group`, the hex codes will be the same.
 
 ```{r}
-display_group_df <- histology_table %>% 
-  dplyr::count(display_group) %>% 
-  dplyr::arrange(n)
-
-knitr::kable(display_group_df)
+cancer_group_df <- data.frame(
+  cancer_group = c("Choroid plexus papilloma",
+                   "Diffuse intrinsic pontine glioma",
+                   "Diffuse midline glioma",
+                   "High-grade glioma astrocytoma",
+                   "Atypical Teratoid Rhabdoid Tumor",
+                   "CNS Embryonal tumor",
+                   "Medulloblastoma",
+                   "Ependymoma",
+                   "Teratoma",
+                   "Ganglioglioma",
+                   "Low-grade glioma astrocytoma",
+                   "Meningioma",
+                   "Ewing sarcoma",
+                   "Dysembryoplastic neuroepithelial tumor",
+                   "Neurofibroma Plexiform",
+                   "Schwannoma",
+                   "Craniopharyngioma"),
+  cancer_group_hex = c("#4d2635",
+                       "#bf0099",
+                       "#ff40d9",
+                       "#ffccf5",
+                       "#4d0d85",
+                       "#b08ccf",
+                       "#a340ff",
+                       "#2200ff",
+                       "#058aff",
+                       "#8c8cff",
+                       "#000080",
+                       "#2db398",
+                       "#9fbf60",
+                       "#614e01",
+                       "#e6ac39",
+                       "#ab7200",
+                       "#b33000"),
+  stringsAsFactors = FALSE
+)
 ```
 
-Make this notebook stop if there are more than 16 histology groups + `Normal`. 
+And again, we'll create a legend with `legend()`
 
 ```{r}
-if (nrow(display_group_df) > 18) {
-  stop("There are more than 18 categories in `display_group`. We may want to re-evaluate the high-level histology groupings")
-}
+plot(NULL, xaxt = "n", yaxt = "n", bty = "n", ylab = "", xlab = "", 
+     xlim = 0:1, ylim = 0:1)
+legend("topleft", 
+       legend = cancer_group_df$cancer_group,
+       col = cancer_group_df$cancer_group_hex,
+       pch = 15, pt.cex = 1.5, cex = 0.75, bty = "n")
+mtext("Cancer Group", at = 0.0625, cex = 1)
 ```
 
-# Make `display_order`
+### Output
 
-Get ranks in order of big to small and make them into a new column in `display_group_df`. 
-We will always want the "normal", "benign", "other_tumor" groups to come last so we will push then to the end of the factor order. 
+We can create a data frame that contains both palettes with a series of left joins, where we will then fill the NA values with a single (gray) hex code per column (`#808080` for `broad_histology`, `#d3d3d3` for `cancer_group`.)
```

Comment:
Got fancy with inserting the colors by variable name here. (Noting that even if you don't do this, the text should be updated to the correct `cancer_group` color.)
```suggestion
We can create a data frame that contains both palettes with a series of left joins, where we will then fill the NA values with a single (gray) hex code per column (`r broad_histology_other_hex` for `broad_histology`, `r cancer_group_other_hex` for `cancer_group`.)
```